analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/LocksReleasedAllPaths.Monitor.Loops.vb

```diff
@@ -6,12 +6,15 @@ Namespace Monitor_Loops
 
         Private Condition As Boolean
         Private Obj As New Object()
+        Private Other As New Object()
 
         Public Sub Method1()
-            Monitor.Enter(Obj)     ' Noncompliant tricky FP, as the execution should always reach number 9, but we don't track that
+            Monitor.Enter(Obj)     ' Compliant for the wrong reason. Should be FP, as the execution should always reach number 9, but we don't track that
             For i As Integer = 0 To 9
                 If i = 9 Then Monitor.Exit(Obj)
             Next
+            Monitor.Enter(Other)    ' FN, exploration stopped after loop
+            If Condition Then Monitor.Exit(Other)
```

Comment:
Every lost Noncompliant should have these two lines too

Reply:
I will add it. Can you elaborate on your reason, though? I am trying to understand why the comment is insufficient for the other cases.

Reply:
To demonstrate, that we stop exploring whatever follows. There are two problems:
* 1st related to the content of the For loop. And that is FN for one reason.
* 2nd related to the fact that we don't explore code after `For` loop

Reply:
The two cases that did not have the additional two lines are not FNs or FPs, though. They are just regular Noncompliant cases.